### DISCUSSION TITLE
Why do you overwrote `global.queueMicrotask`?

### DISCUSSION THREAD

**Discussion by tyson90** [software-mansion/react-native-reanimated#4716] [Q&A]
I just updated RN Reanimated to v3.3.0 (and other libraries, including RN). In my app I use queueMicrotask in several places. After hours of invastigating why some functionality in my app now is broken, I discovered that you overwrote `global.queueMicrotask` in `/lib/module/reanimated2/threads.js` file. My question is why? And why you didn't mention it as breaking change?

**Comment by piaskowyk** [comment]
> My question is why? And why you didn't mention it as breaking change?

We don't overwrite React's queueMicrotask function. We just create own function on our own runtime on UI thread. Method `setupMicrotasks` is called inside `runOnUIImmediately` - https://github.com/software-mansion/react-native-reanimated/blob/main/src/reanimated2/initializers.ts#L187
Your issue probably is caused by old cache. Try to run `yarn start --clear-cache`
